kotlinpoet/src/main/java/com/squareup/kotlinpoet/FileSpec.kt

```diff
@@ -88,11 +91,15 @@ public class FileSpec private constructor(
 
     val outputPath = outputDirectory.resolve("$name.$extension")
     OutputStreamWriter(Files.newOutputStream(outputPath), UTF_8).use { writer -> writeTo(writer) }
+    return outputPath
   }
 
-  /** Writes this to `directory` as UTF-8 using the standard directory structure.  */
+  /**
+   * Writes this to `directory` as UTF-8 using the standard directory structure and returns the path
+   * of the output file.
+   */
   @Throws(IOException::class)
-  public fun writeTo(directory: File): Unit = writeTo(directory.toPath())
+  public fun writeTo(directory: File): File = writeTo(directory.toPath()).toFile()
```

Comment:
For reasons I don't understand, Path.toFile() may throw UnsupportedOperationException so technically this could be a regression even if nobody wants to know the return value. But JavaPoet does the same [here](https://github.com/square/javapoet/blob/master/src/main/java/com/squareup/javapoet/JavaFile.java#L159-L160), and since we already had a File to begin with, I hope it should be fine.